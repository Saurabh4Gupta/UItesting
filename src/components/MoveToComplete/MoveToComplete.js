import React from 'react';
import { Modal, Button } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
// import { useHistory, useLocation } from 'react-router';
// import Toast from '@dentsu-ui/components/dist/cjs/components/Toast';
// import { dataFieldCms as PageContent } from '../../cms';


const MoveToComplete = (props) => {
  const { cmsData, setModalOpen, modalOpen, handleMoveToComplete } = props;
  const onCloseModal = () => {
    setModalOpen(false);
  };

  return (
    <>
      <Modal
        isFullHeight={false}
        width="500px"
        isOpen={modalOpen}
        onClose={onCloseModal}
      >
        <Modal.Header
          hasCloseButton
          title={cmsData.moveToCompleteConfirmation}
        />
        <Modal.Body>
          <p>{cmsData.moveToCompleteDescription}</p>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={onCloseModal}>
            {cmsData.deleteButtonCancelMessage}
          </Button>
          <Button onClick={handleMoveToComplete}>{cmsData.yesContinue}</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};
MoveToComplete.propTypes = {
  setModalOpen: PropTypes.func,
  modalOpen: PropTypes.bool,
  cmsData: PropTypes.object,
  handleMoveToComplete: PropTypes.func,
};

MoveToComplete.defaultProps = {
  setModalOpen: () => { },
  modalOpen: false,
  cmsData: {},
  handleMoveToComplete: () => { },
};
export default MoveToComplete;
