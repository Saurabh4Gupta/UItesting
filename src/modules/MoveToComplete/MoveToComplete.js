import React from 'react';
import { Modal, Button } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
import { useHistory, useLocation } from 'react-router';
import Toast from '@dentsu-ui/components/dist/cjs/components/Toast';
import { dataFieldCms as PageContent } from '../../cms';


const MoveToComplete = (props) => {
  const { cmsData, setModalOpen, modalOpen, requestId } = props;
  const history = useHistory();
  const location = useLocation();

  const query = new URLSearchParams(location.search);

  const clientCode = query.get('client_code');

  const onCloseModal = () => {
    setModalOpen(false);
  };

  const onSubmit = (id) => {
    setModalOpen(false);
    console.log(id);
    const queryString = `client_code=${clientCode}`;
    history.push({
      pathname: '/datafield',
      search: `?${queryString}`,
      state: { requestId: id  },
    });
    const toast = new Toast();

    return toast({
      title: '',
      content: PageContent.toastMovedToComplete,
      status: 'success',
    });
  };
  return (
    <>
      <Modal
        isFullHeight={false}
        width="500px"
        isOpen={modalOpen}
        onClose={onCloseModal}
      >
        <Modal.Header
          hasCloseButton
          title={cmsData.moveToCompleteConfirmation}
        />
        <Modal.Body>
          <p>{cmsData.moveToCompleteDescription}</p>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={onCloseModal}>
            {cmsData.deleteButtonCancelMessage}
          </Button>
          <Button onClick={() => onSubmit(requestId)}>
            {cmsData.yesContinue}
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};
MoveToComplete.propTypes = {
  setModalOpen: PropTypes.func,
  modalOpen: PropTypes.bool,
  cmsData: PropTypes.object,
  requestId: PropTypes.string,
};

MoveToComplete.defaultProps = {
  setModalOpen: () => {},
  modalOpen: false,
  cmsData: {},
  requestId: PropTypes.string,
};
export default MoveToComplete;
