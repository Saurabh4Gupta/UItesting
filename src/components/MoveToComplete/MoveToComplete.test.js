import React from 'react';
import { shallow } from 'enzyme';
import MoveToComplete from './MoveToComplete';

describe('Success test cases for: MoveToComplete component', () => {
  let wrapper;
  const props = {
    setModalOpen: jest.fn(),
  }

  describe('Snapshot test for MoveToComplete component', () => {
    beforeEach(() => {
      wrapper = shallow(<MoveToComplete {...props} />);
    });

    it('Test to match the ForwardRef', async (done) => {
      expect(wrapper).toMatchSnapshot();
      done();
    });

    it('Test to match the snapsot', async () => {
      const fn = wrapper.find('ForwardRef').at(0).prop('onClose');
      expect(fn).toBeInstanceOf(Function);
      fn();
    });

    it('Renders correctly', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });
});

describe('Failure test cases for: Duplicate data page', () => { });
