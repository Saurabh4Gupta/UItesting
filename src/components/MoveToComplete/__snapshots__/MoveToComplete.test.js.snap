// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success test cases for: MoveToComplete component Snapshot test for MoveToComplete component Test to match the ForwardRef 1`] = `
<Fragment>
  <ForwardRef
    isFullHeight={false}
    isOpen={false}
    onClose={[Function]}
    width="500px"
  >
    <ForwardRef
      hasCloseButton={true}
    />
    <ForwardRef>
      <p />
    </ForwardRef>
    <ForwardRef>
      <ForwardRef
        onClick={[Function]}
        variant="secondary"
      />
      <ForwardRef
        onClick={[Function]}
      />
    </ForwardRef>
  </ForwardRef>
</Fragment>
`;
